loader.go

```diff
@@ -108,4 +119,47 @@ func GetSingleton[t any]() (t, bool) {
 		return x, ok
 	}
 	return v.(t), ok
+}
+
+func GetConfig[T any](name string) (*T, bool) {
+
+	if configPath == "" {
+		execPath, _ := os.Executable()
+		execDir := filepath.Dir(execPath)
+		configPath = filepath.Join(execDir, "pluginconfig")
+		log.Warn("no plugin config path set, config path set to default")
+	}
+
+	pathInfo, err := os.Stat(configPath)
+	if !pathInfo.IsDir() {
+		log.Error("the provided plugin config path is not a directory, config path set to default")
+	}
+	if os.IsNotExist(err) {
+		log.Error("the provided plugin config path does not exist, config path set to default")
+	} else if err != nil {
+		log.Error("error while checking the provided plugin config path, config path set to default", "err", err)
+	}
+	
+	c := new(T)
+
+	file := configPath + "/" + name + ".yaml"
```

Comment:
Rather than `configPath + "/" + name + ".yaml"` I'd rather see `filepath.Join(configPath, name +".yaml")`, as this will be more cross platform. Old PluGeth wouldn't have worked on windows because of limitations on dynamic imports, but PluGeth2 theoretically could, and I'd rather not break that over how we handle imports.